pkg/controllers/csi/metadata/sqlite_gorm_client.go

```diff
@@ -11,39 +11,46 @@ import (
 	"gorm.io/gorm/logger"
 )
 
-type DBAccess interface {
-	SchemaMigration(ctx context.Context) error
-
-	CreateTenantConfig(ctx context.Context, tenantConfig *TenantConfig) error
-	UpdateTenantConfig(ctx context.Context, tenantConfig *TenantConfig) error
-	DeleteTenantConfig(ctx context.Context, tenantConfig *TenantConfig) error
-	ReadTenantConfigByTenantUUID(ctx context.Context, uid string) (*TenantConfig, error)
-
-	CreateCodeModule(ctx context.Context, codeModule *CodeModule) error
-	ReadCodeModuleByVersion(ctx context.Context, version string) (*CodeModule, error)
-	DeleteCodeModule(ctx context.Context, codeModule *CodeModule) error
-
-	CreateOSMount(ctx context.Context, osMount *OSMount) error
-	UpdateOSMount(ctx context.Context, osMount *OSMount) error
-	DeleteOSMount(ctx context.Context, osMount *OSMount) error
-	RestoreOSMount(ctx context.Context, osMount *OSMount) error
-	ReadOSMountByTenantUUID(ctx context.Context, tenantUUID string) (*OSMount, error)
-
-	CreateAppMount(ctx context.Context, appMount *AppMount) error
-	UpdateAppMount(ctx context.Context, appMount *AppMount) error
-	DeleteAppMount(ctx context.Context, appMount *AppMount) error
-	ReadAppMountByVolumeMetaID(ctx context.Context, volumeMetaID string) (*AppMount, error)
-	ReadAppMounts(ctx context.Context) ([]AppMount, error)
+type GormAccess interface {
+	SchemaMigration() error
+
+	ReadTenantConfig(tenantConfig TenantConfig) (*TenantConfig, error)
+	ReadCodeModule(codeModule CodeModule) (*CodeModule, error)
+	ReadOSMount(osMount OSMount) (*OSMount, error)
+	ReadAppMount(appMount AppMount) (*AppMount, error)
+
+	ReadTenantConfigs() ([]TenantConfig, error)
+	ReadCodeModules() ([]CodeModule, error)
+	ReadOSMounts() ([]OSMount, error)
+	ReadAppMounts() ([]AppMount, error)
+	ReadVolumeMetas() ([]VolumeMeta, error)
+
+	CreateTenantConfig(tenantConfig *TenantConfig) error
+	CreateCodeModule(codeModule *CodeModule) error
+	CreateOSMount(osMount *OSMount) error
+	CreateAppMount(appMount *AppMount) error
+
+	UpdateTenantConfig(tenantConfig *TenantConfig) error
+	UpdateOSMount(osMount *OSMount) error
+	UpdateAppMount(appMount *AppMount) error
+
+	DeleteTenantConfig(tenantConfig *TenantConfig, cascade bool) error
+	DeleteCodeModule(codeModule *CodeModule) error
+	DeleteOSMount(osMount *OSMount) error
+	DeleteAppMount(appMount *AppMount) error
+
+	IsCodeModuleOrphaned(codeModule *CodeModule) (bool, error)
 }
 
-type DBConn struct {
-	db *gorm.DB
+type GormConn struct {
+	ctx context.Context
```

Comment:
I was under the impression, that storing Contexts is frowned upon- is there a special circumstance here, that makes this preferable compared to passing it on call?

Reply:
resolved- as we discussed it in standup